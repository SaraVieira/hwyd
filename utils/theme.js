import {
  /* colors */
  APPLE_VALLEY,
  BREWED_MUSTARD,
  CREAMY_PEACH,
  OLD_GERANIUM,
  PENCIL_LEAD,
  ROGUE_PINK,
  ROSY_HIGHLIGHT,
  SOFT_BLUE,
  SQUEAKY,
  WHITE,

  /* fonts */
  BOLD_FONT,
  ITALIC_FONT,
  REGULAR_FONT,
  SUBTITLE_FONT,
  TITLE_FONT,
} from './constants';

export default {
  colors: {
    appleValley: APPLE_VALLEY,
    brewedMustard: BREWED_MUSTARD,
    creamyPeach: CREAMY_PEACH,
    oldGeranium: OLD_GERANIUM,
    pencilLead: PENCIL_LEAD,
    roguePink: ROGUE_PINK,
    rosyHighlight: ROSY_HIGHLIGHT,
    softBlue: SOFT_BLUE,
    squeaky: SQUEAKY,
    white: WHITE,
  },
  fonts: {
    bold: BOLD_FONT,
    italic: ITALIC_FONT,
    regular: REGULAR_FONT,
    subtitle: SUBTITLE_FONT,
    title: TITLE_FONT,
  },
};
