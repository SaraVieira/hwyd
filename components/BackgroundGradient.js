/* @flow */

import React, { Component } from 'react';
import { LinearGradient } from 'expo';
import styled from 'styled-components';

import { ROGUE_PINK, ROSY_HIGHLIGHT } from '../utils/constants';

const Background = styled(LinearGradient)`
  bottom: 0;
  left: 0;
  position: absolute;
  right: 0;
  top: 0;
`;

export default class BackgroundGradient extends Component<*, *> {
  render = () => (
    <Background colors={[ROGUE_PINK, ROSY_HIGHLIGHT]} {...this.props} />
  );
}
