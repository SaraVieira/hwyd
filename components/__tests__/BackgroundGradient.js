import React from 'react';
import { ThemeProvider } from 'styled-components';
import { shallow } from 'enzyme';
import 'jest-styled-components';

import BackgroundGradient from '../BackgroundGradient';
import theme from '../../utils/theme';

describe('<BackgroundGradient />', () => {
  it('renders a linear gradient background', () => {
    const tree = shallow(
      <ThemeProvider theme={theme}>
        <BackgroundGradient />
      </ThemeProvider>,
    );
    const root = tree.find(BackgroundGradient).shallow();
    expect(root).toMatchSnapshot();
  });
});
