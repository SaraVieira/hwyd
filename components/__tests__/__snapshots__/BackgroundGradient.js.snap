// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<BackgroundGradient /> renders a linear gradient background 1`] = `
<Styled(LinearGradient)
  colors={
    Array [
      "#f8a5c2",
      "#f7d794",
    ]
  }
/>
`;
